### DISCUSSION TITLE
Correct way of authentication in C#

### DISCUSSION THREAD

**Discussion by JohnCido** [Palakis/obs-websocket#914] [Q&A] [answered]
I've tried to implement the authentication in my Windows App SDK app. But the auth seems always fail. I can see my client appears on the obs-websocket settings window.
This is my authentication string generation:
```C#
var secret = Helpers.Base64Encode(Helpers.ComputeSha256Hash($"{Password}{auth.salt}"));
var pass = Helpers.Base64Encode(Helpers.ComputeSha256Hash($"{secret}{auth.challenge}"));
data.authentication = pass;
``` 
All the JSON are correctly deserialized. The password is correct too.

And these are the helper functions:
```C#
private static byte[] GetUtf8Bytes(string input)
{
    return Encoding.UTF8.GetBytes(input);
}

public static string Base64Encode(string input)
{
    return Convert.ToBase64String(GetUtf8Bytes(input));
}

public static string ComputeSha256Hash(string input)
{
    // Create a SHA256
    using SHA256 sha256Hash = SHA256.Create();
    // ComputeHash - returns byte array
    byte[] bytes = sha256Hash.ComputeHash(GetUtf8Bytes(input));

    // Convert byte array to a input
    StringBuilder builder = new();
    foreach (var b in bytes)
    {
        builder.Append(b.ToString("x2"));
    }
    return builder.ToString();
}
```
Is there any mistake I made during this progress or it's a problem rooted somewhere else?

**Comment by tt2468** [comment, marked as answer]
I'm not a c# dev, but here's how obs-websocket-dotnet handles their authentication: https://github.com/BarRaider/obs-websocket-dotnet/blob/v5-dev/obs-websocket-dotnet/OBSWebsocket.cs#L548

**Reply by JohnCido** [reply]
Thanks for giving this reference! I've fixed it in my app now.
